numpy/distutils/fcompiler/environment.py

```diff
@@ -49,10 +49,24 @@ def get(self, name, default=None):
         return var
 
     def _get_var(self, name, conf_desc):
-        hook, envvar, confvar, convert = conf_desc
+        hook, envvar, confvar, convert, append = conf_desc
         var = self._hook_handler(name, hook)
         if envvar is not None:
-            var = os.environ.get(envvar, var)
+            envvar_contents = os.environ.get(envvar)
+            if envvar_contents is not None:
+                if convert is not None:
+                    envvar_contents = convert(envvar_contents)
+                else:
+                    envvar_contents = [envvar_contents]
+                if append and os.environ.get('NPY_DISTUTILS_APPEND_FLAGS', '0') == '1':
+                    if var is None:
+                        var = ''
+                    try:
+                        var = ' '.join(var + envvar_contents)
+                    except TypeError:
+                        var = ' '.join([var] + envvar_contents)
```

Comment:
Can you add a comment explaining why the try catch is needed? Specifically, give an example of when `var` is a list and is a string.

Why was this string / list uncertainty not a problem before?

Reply:
This is unnecessary after https://github.com/numpy/numpy/pull/11525/commits/65687ff01c932b53d641fbe91b8d945d1a550744#diff-794348013d3654034275e0453541e969L117